xm-commons-lep-groovy/src/main/java/com/icthh/xm/commons/lep/groovy/GroovyFileParser.java

```diff
@@ -0,0 +1,261 @@
+package com.icthh.xm.commons.lep.groovy;
+
+import groovyjarjarantlr.RecognitionException;
+import groovyjarjarantlr.TokenStreamException;
+import groovyjarjarantlr.collections.AST;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.codehaus.groovy.antlr.AntlrASTProcessor;
+import org.codehaus.groovy.antlr.GroovySourceAST;
+import org.codehaus.groovy.antlr.SourceBuffer;
+import org.codehaus.groovy.antlr.UnicodeEscapingReader;
+import org.codehaus.groovy.antlr.parser.GroovyLexer;
+import org.codehaus.groovy.antlr.parser.GroovyRecognizer;
+import org.codehaus.groovy.antlr.treewalker.SourceCodeTraversal;
+import org.codehaus.groovy.antlr.treewalker.Visitor;
+import org.codehaus.groovy.antlr.treewalker.VisitorAdapter;
+
+import java.io.StringReader;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.codehaus.groovy.antlr.parser.GroovyTokenTypes.IDENT;
+import static org.codehaus.groovy.antlr.parser.GroovyTokenTypes.LITERAL_static;
+import static org.codehaus.groovy.antlr.parser.GroovyTokenTypes.MODIFIERS;
+
+@Slf4j
+public class GroovyFileParser {
+
+    @SneakyThrows
+    public GroovyFileMetadata getFileMetaData(String source) {
+
+        try {
+            GroovyFileMetadata metadata = new GroovyFileMetadata();
+            parseGroovy(source, new VisitorAdapter() {
```

Comment:
Suggest creation of at least inner class for `VisitorAdapter` implementation. Or even top level class.

Reply:
what wrong with inner classes?

Reply:
nothing. But for me this logic deserves to be in separate class not just in anonymous.
for example even in case exception occurs then you will see normal class in stacktrace which could be easily navigable in IDE.